src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1123,6 +1273,54 @@ SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
             }
         }
 
+
+        private static readonly byte[] TimedCounterPrefixBuffer = Encoding.UTF8.GetBytes(TimedCounterPrefix);
+        private static readonly byte[] TimedCounterPositivePrefixBuffer = Encoding.UTF8.GetBytes(TimedCounterPositivePrefix);
+        private const string TimedCounterPrefix = "TC:";
+        private const string TimedCounterPositivePrefix = TimedCounterPrefix + "INC-";
+        private const string TimedCounterNegativePrefix = TimedCounterPrefix + "DEC-";
+        public string IncrementTimestamp(
+            DocumentsOperationContext context,
+            string documentId,
+            string collection,
+            string name,
+            IEnumerable<TimeSeriesOperation.IncrementOperation> toIncrement)
+        {
+#if DEBUG
+            DateTime prevTime = DateTime.MinValue;
+#endif
+
+            var holder = new SingleResult();
+
+            return AppendTimestamp(context, documentId, collection, name, toIncrement.Select(ToResult));
+
+
+            SingleResult ToResult(TimeSeriesOperation.IncrementOperation element)
+            {
+#if DEBUG
+                if (element.Timestamp <= prevTime)
+                {
+                    throw new InvalidDataException(
+                        $"The order of increment operations must be sequential, but got previous operation {prevTime} and the current: {element.Timestamp}");
+                }
+
+                prevTime = element.Timestamp;
+#endif
+                var sign = Math.Sign(element.Values[0]);
+                for (int i = 1; i < element.Values.Length; i++)
+                {
+                    if (sign != Math.Sign(element.Values[i]))
```

Comment:
Hmmm.. interesting case.. not sure if we want to go there, but we can split it into positive/negative entries..
e.g. `Inc([1,-1])` will be resulted as `Inc([1,0])` and `Inc([0,-1])`

Note that `Sign` can also return `0`, but `Inc([1,0])` should be valid it think.. (please add a test for those cases)

Reply:
Done